src/main/java/opendcs/dao/DaoBase.java

```diff
@@ -99,6 +100,20 @@ public DaoBase(DatabaseConnectionOwner tsdb, String module, Connection con)
         this.myCon = con;
     }
 
+    /**
+     * When used within the transaction block of another Dao allow this to assume the same connection.
+     * @param other
+     * @throws IllegalStateException if the other Dao does not already have a connection open this operation is not valid.
+     */
+    public void  inTransactionOf(DaoBase other) throws IllegalStateException
+    {
+        if (other.myCon == null)
+        {
+            throw new IllegalStateException("Provided DAO does not currently have a valid connection that would be in a transaction.");
+        }
+        this.setManualConnection(other.myCon);
```

Comment:
When this/DaoBase assumes connection in another Dao, by calling setManualConnection(other.myCon).

Should there be cleanup when finished, to get back to original connection?

Reply:
I'm going to have to think about that, The way I've been using them it doesn't matter as they only live during the transaction block itself. But there could be a reasonable way to allow that without any specific work on the end user.

Reply:
There's not really a good way to handle this, I put a not in java doc that the Dao should be considered invalid after calling this.